python_ta/typecheck/base.py

```diff
@@ -448,8 +448,14 @@ def parse_annotations(node, class_tvars=None):
     """Return a type specified by the type annotations for a node."""
     if isinstance(node, astroid.FunctionDef):
         arg_types = []
+        arg_annotation_list = list(zip(node.args.args, node.args.annotations))
+        # Special case; Instance method definition based on syntactical conventions.
         if class_tvars is None or not isinstance(node.parent, astroid.ClassDef):
-            self_type = None
+            if isinstance(node.parent, astroid.ClassDef) and getattr(
```

Comment:
The nested `if` structure here isn't great. Refactor this code so that the three cases are explicit in one `if-elif-else` chain. Also remove `arg_annotation_list` - its only use here does not warrant its existence.